### DISCUSSION TITLE
PMP::clip(mesh, plane) crashes the session when using Valgrind

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7193] [Q&A]
Hello,

I checked my R package with Valgrind and when clipping a mesh to a plane with `PMP::clip`, there's a crash. You can find the Valgrind report and a reproducible code [in this gist](https://gist.github.com/stla/27c8251b7a22909f88da0bd174ceebe7). I tried to put some `std::cout` in a corefinement visitor and then I observed that `after_face_copy()` is never triggered. The actions `before_subface_creation` and `after_subface_created` are triggered.

I always use `Epeck` and usually when I have a crash with Valgrind, I also have the same crash if I replace `Epeck` with `Epick`. I don't think this is the case here, but I will try.

**Comment by stla** [comment]
> I always use `Epeck` and usually when I have a crash with Valgrind, I also have the same crash if I replace `Epeck` with `Epick`. I don't think this is the case here, but I will try.

I confirm: no crash with `Epick`.

**Comment by sloriot** [comment]
Valgrind does not like rounding mode changes done by CGAL. See https://github.com/CGAL/cgal/issues/3094

**Comment by stla** [comment]
Ok. But this error is different to the ones I already encountered with Valgrind. 

Just checking another example of `clip(mesh, plane)` and crash as well, but not the same.

The other clipping functions, `clip(mesh1, mesh2)` and `clip(mesh, isocuboid)`, work fine with Valgrind.

**Comment by sloriot** [comment]
Except if you compile your exe with the option to disable rounding mode changes, Valgrind is not expected to work with CGAL. If you can reproduce an error without Valgrind I can have a look otherwise it is most probably an usual issue with Valgrind.
